app/src/components/market/profile/closed_market_detail.tsx

```diff
@@ -109,13 +119,14 @@ export const ClosedMarketDetailWrapper = (props: Props) => {
 
   const redeem = async () => {
     try {
+      setStatus(Status.Loading)
       if (!isConditionResolved) {
         setMessage('Resolving condition...')
-        await resolveCondition()
+        // Balances length is the number of outcomes
+        await oracle.resolveCondition(questionId, balances.length)
       }
-      setMessage('Redeem payout...')
-      setStatus(Status.Loading)
 
+      setMessage('Redeem payout...')
```

Comment:
Should we also add a `setMessage('Resolving condition...')` line before the if? Or maybe just move this line before the if?

Reply:
I think the message should be added inside the `if` , is only needed when the condition is not resolved. Mmmm you make me think why we need to put this message before the `if` , do you have any concern about ? Please let me know, your thoughts are really appreciated. This is how it looks now...

![Screenshot_20191219_103405](https://user-images.githubusercontent.com/1144028/71177811-92dbee80-224b-11ea-8d66-6498966d9332.png)

I tested and works pretty good

Reply:
Yeah, inside the `if` is better. And it's just because the user will get 2 transactions, and so I think it's better to explain what each one is doing.